couchbase/tests/test_couchbase.py

```diff
@@ -194,7 +187,47 @@ def _assert_bucket_metrics(aggregator, tags, device=None):
 def _assert_stats(aggregator, node_tags, device=None):
     for mname in NODE_STATS:
         aggregator.assert_metric('couchbase.by_node.{}'.format(mname), tags=node_tags, count=1, device=device)
-
     # Assert 'couchbase.' metrics
     for mname in TOTAL_STATS:
         aggregator.assert_metric('couchbase.{}'.format(mname), tags=CHECK_TAGS, count=1)
+
+
+@pytest.mark.skipif(COUCHBASE_MAJOR_VERSION != 7, reason='Index metrics are only available for Couchbase 7+')
+@pytest.mark.integration
+@pytest.mark.usefixtures("dd_environment")
+def test_index_stats_metrics(aggregator, dd_run_check, instance_index_stats, couchbase_container_ip):
+    """
+    Test Index Statistics metrics (prefixed "couchbase.index." and "couchbase.indexer.")
+    """
+    couchbase = Couchbase('couchbase', {}, [instance_index_stats])
+    dd_run_check(couchbase)
+    for mname in INDEX_STATS_INDEXER_METRICS:
+        aggregator.assert_metric(mname, tags=CHECK_TAGS)
+
+    for mname in INDEX_STATS_INDEX_METRICS:
+        aggregator.assert_metric(mname, tags=INDEX_STATS_TAGS)
```

Comment:
```suggestion
        aggregator.assert_metric(mname, type=..., tags=INDEX_STATS_TAGS)
```
nit- since now there are some metrics that are counts can we also assert the metric type in this test?

Reply:
Not sure if I done this correctly. I attempted it in my latest commit. I'm guessing metric_type 0 is a gauge and metric_type 3 is a count?

Reply:
That works, but a more readable way is to use `metric_type=aggregator.GAUGE` which uses the same values from [here](https://github.com/DataDog/integrations-core/blob/master/datadog_checks_base/datadog_checks/base/stubs/aggregator.py#L71)

Reply:
Ahh that's where that is from.